### PR TITLE
Fixing incident alerts and artifacts is not populated  

### PR DESCRIPTION
**Pull request by maimorag**
## Status
- [x] In Progress

## Related Issues
fixes:[ link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-35287)
https://jira-dc.paloaltonetworks.com/browse/XSUP-35782

## Description
- Fixing incident alerts is not populated on new incidents when bringing them into XSOAR.
- Fixing ALERTS_LIMIT_PER_INCIDENTS is not set when docker has a timeout and incident not_created_incidents are left by adding it to the last run dictionary.
- Fixing:
```
File "<Cortex XDR - IR>", line <XX_REPLACED>86, in <lambda>
 TypeError: list indices must be integers or slices, not str
```

### CODE REVIEW

**Comment by maimorag** [comment]
commit [a053695](https://github.com/demisto/content/pull/33558/commits/a053695fd373486aac21c56d5ef645b98ed67d94)
contains a failed unit test test_sort_all_list_incident_fields_case_get_multiple_incidents_extra_data_format that reproduced the bug

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 1]
```diff
@@ -522,6 +522,24 @@ def get_last_mirrored_in_time(args):
     return last_mirrored_in_timestamp
 
 
+def sort_incident_data(raw_incident):
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:525]: Lets add docstring explaining what the method is doing.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 2]
```diff
@@ -522,6 +522,24 @@ def get_last_mirrored_in_time(args):
...
+    incident = raw_incident.get('incident', {})
+    raw_alerts = raw_incident.get('alerts', {}).get('data', None)
+    file_artifacts = raw_incident.get('file_artifacts', {}).get('data')
+    network_artifacts = raw_incident.get('network_artifacts', {}).get('data')
+    context_alerts = []
+    if raw_alerts:
+        context_alerts = clear_trailing_whitespace(raw_alerts)
+        for alert in context_alerts:
+            alert['host_ip_list'] = alert.get('host_ip').split(',') if alert.get('host_ip') else []
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:534]:
```suggestion
    context_alerts = clear_trailing_whitespace(raw_alerts)
    if context_alerts:
        for alert in context_alerts:
            alert['host_ip_list'] = alert.get('host_ip').split(',') if alert.get('host_ip') else []
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 3]
```diff
@@ -1000,16 +1009,15 @@ def fetch_incidents(client, first_fetch_time, integration_instance, last_run: di
         count_incidents = 0
 
         for raw_incident in raw_incidents:
-            incident_data: dict[str, Any] = raw_incident.get('incident') or raw_incident
+            incident_data: dict[str, Any] = sort_incident_data(raw_incident) if raw_incident.get('incident') else raw_incident
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1012]: What happens when we have no incident as key in the raw incident? The dict keys are directly the incident fields?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 4]
```diff
@@ -1021,14 +1029,11 @@ def fetch_incidents(client, first_fetch_time, integration_instance, last_run: di
                 'occurred': occurred,
                 'rawJSON': json.dumps(incident_data),
             }
-
             if demisto.params().get('sync_owners') and incident_data.get('assigned_user_mail'):
                 incident['owner'] = demisto.findUser(email=incident_data.get('assigned_user_mail')).get('username')
-
             # Update last run and add incident if the incident is newer than last fetch
-            if incident_data['creation_time'] > last_fetch:
+            if incident_data.get('creation_time', 0) > last_fetch:
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1035]: What is the format of last fetch? what 0 in timestamp/datetime supposed to do? Did you test this case?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py [hunk 5]
```diff
@@ -1267,3 +1267,24 @@ def test_get_incident_extra_data_incident_not_exist(mocker):
...
+    Then
+        - Verify that alerts and artifacts are found.
+    """
+    from CortexXDRIR import sort_incident_data, sort_all_list_incident_fields
+    incident_case_get_multiple_incidents_extra_data = load_test_data('./test_data/get_multiple_incidents_extra_data.json')\
+        .get('reply').get('incidents')[0]
+    incident_data = sort_incident_data(incident_case_get_multiple_incidents_extra_data)
+    sort_all_list_incident_fields(incident_data)
+    assert incident_data.get('alerts')
+    assert incident_data.get('incident_sources')
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:1288]: Lets test the values we are expecting to get

File: Packs/CortexXDR/Integrations/CortexXDRIR/test_data/get_incident_extra_data_new_status.json [hunk 6]
```diff
@@ -1,5 +1,8 @@
 {
     "reply": {
+        "incidents":
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/test_data/get_incident_extra_data_new_status.json:3]: What are those changes meant to represent? the response from the API is different than what we thought?

File: Packs/CortexXDR/ReleaseNotes/6_1_27.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+
+- Fixed an issue where ***alerts***, ***network_artifacts***, ***file_artifacts*** is not being populated on new incidents when bringing them into XSOAR.
```
Comment (eyalpalo) [inline Packs/CortexXDR/ReleaseNotes/6_1_27.md:6]: Add rn regarding other fixes in this version

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py [hunk 8]
```diff
@@ -94,8 +94,8 @@ def test_fetch_incidents_filtered_by_status(requests_mock, mocker):
     client = Client(
         base_url=f'{XDR_URL}/public_api/v1', verify=False, timeout=120, proxy=False)
     incident_extra_data_under_investigation = load_test_data('./test_data/get_incident_extra_data_host_id_array.json')\
-        .get('reply', {}).get('incidents')[0]
-    incident_extra_data_new = load_test_data('./test_data/get_incident_extra_data_new_status.json').get('reply')
+        .get('reply', {}).get('incidents')
+    incident_extra_data_new = load_test_data('./test_data/get_incident_extra_data_new_status.json').get('reply').get('incidents')
```
Comment (eyalpalo) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:98]: What are we testing here? you removed all information from this test data file?

**Review by eyalpalo — CHANGES_REQUESTED** [review]
We are fixing here 2 bugs. I would like us to have at least 2 tests that are testing the changes that were made as a result. Also, tests to test the changes in the code.

File: Packs/CortexXDR/Integrations/CortexXDRIR/test_data/get_incident_extra_data_new_status.json (hunk 6 above)
Comment (maimorag) [reply Packs/CortexXDR/Integrations/CortexXDRIR/test_data/get_incident_extra_data_new_status.json:3]: I fixed the format, it suppose to be as mentioned here: 
https://cortex-panw.stoplight.io/docs/cortex-xdr/a61eadc13dd54-get-all-incidents

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 3 above)
Comment (maimorag) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1012]: yes

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py (hunk 8 above)
Comment (maimorag) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:98]: fetching incidents with different statues. 
The information wasn't removed, I fixed the structure accordingly

File: Packs/CortexXDR/ReleaseNotes/6_1_27.md [hunk 9]
```diff
@@ -4,3 +4,5 @@
 ##### Palo Alto Networks Cortex XDR - Investigation and Response
 
 - Fixed an issue where ***alerts***, ***network_artifacts***, ***file_artifacts*** is not being populated on new incidents when bringing them into XSOAR.
+- Fixed an issue which alert_limit_incident isn't set when docker has a timeout and incidents were not added yet.
```
Comment (eyalpalo) [inline Packs/CortexXDR/ReleaseNotes/6_1_27.md:7]:
```suggestion
- Fixed an issue where the alert count was reset after docker timeout.
```

File: Packs/CortexXDR/ReleaseNotes/6_1_27.md [hunk 10]
```diff
@@ -4,3 +4,5 @@
...
+- Fixed an parsing the response's format differed from the origin after improving implementation.
```
Comment (eyalpalo) [inline Packs/CortexXDR/ReleaseNotes/6_1_27.md:8]:
```suggestion
- Fixed a bug where the response from the API was not parsed correctly.
```

**Review by eyalpalo — APPROVED** [review]

**Comment by maimorag** [comment]
Test playbooks fail on:
Test Playbook - Cortex XDR - Retrieve File by sha256 CIAC-9932
Test Playbook - Cortex XDR Malware - Incident Enrichment CIAC-9931

**Pull request merged**